app_dart/bin/generate_jspb.dart

```diff
@@ -52,20 +58,46 @@ String getLocalConfigContent(String path) {
 }
 
 Future<void> main(List<String> args) async {
-  if (args.length != 1 && args.length != 2) {
+  if (args.length != 1 && args.length != 3) {
     print('generate_jspb.dart \$local_ci_yaml');
-    print('generate_jspb.dart \$repo \$sha');
+    print('generate_jspb.dart \$repo \$sha \$branch');
     exit(1);
   }
   String configContent;
-  if (args.length == 2) {
+  if (args.length == 3) {
     configContent = await getRemoteConfigContent(args[0], args[1]);
   } else {
     configContent = getLocalConfigContent(args[0]);
   }
-  final YamlMap configYaml = loadYaml(configContent) as YamlMap;
-  // There's an assumption that we're only generating builder configs from commits that
-  // have already landed with validation. Otherwise, this will fail.
-  final pb.SchedulerConfig schedulerConfig = schedulerConfigFromYaml(configYaml);
+
+  pb.SchedulerConfig schedulerConfig;
+  if (args.length == 3) {
+    final FakeScheduler scheduler = FakeScheduler(config: FakeConfig());
+    Commit currentCommit = generateCommit(1, repo: args[0], sha: args[1], branch: args[2]);
+
+    // FOR REVIEW:
+    // requiring branch information from user as well to check whether current branch is a release branch.
+    if (args[2] == Config.defaultBranch(RepositorySlug('flutter', args[0]))) {
+      // FOR REVIEW:
+      // supply 0 in generateCommit, to singal that we generate a commit with an empty sha
+      Commit totCommit =
```

Comment:
Thanks for the reply! 

Umm the reason I am using a fake scheduler is that, currently [getCiYaml] function is under [Scheduler] class. To access [getCiYaml] in tooling files, I am avoiding creating a real scheduler which requires cache, luckBuildService etc., which might also need fakes values to get initialized. And based on previous learnings I did not to convert [getCiYaml] to a static function. There might be a better way, but eventually I decided to use a fakeScheduler to give tooling files access to [getCiYaml].

For the validate_scheduler_config.dart file, it is also a tooling but from the name of it I am assuming it would need to trigger the actual validation tests by calling upon getCiYaml. Currently I am avoiding creating a real scheduler (which needs required parameters such as cache, luciBuildServices etc.), and instead using fakeScheduler.getRealCiYaml to run the validation check. The getCiYaml function under fakeScheduler does not run the underlying validation checks, and I am using it for other files such as update_existing_flaky_issues. 

umm I guess I was not very sure how to give tooling files access to [scheduelr.getCiYaml] without creating fakes or static functions.

Reply:
umm just in case it gets buried in other emails. Would be great if you can take a look the above comment from yesterday as well. Thank you!

Reply:
Again, I want to emphasize never to use fakes or to create dummy values to work around issues. It's a code smell / red flag, and we can talk about it more in our next 1:1

We should leave the logic as is (downloading the yaml map and pass it ourselves). This tooling does not need the extra caching or datastore logic that `Scheduler` introduces (and it wouldn't work on LUCI bots). So instead we should directly use `CiYaml`

`validate_scheduler_config.dart` is a local way to run the validation, but it's not used in CI

Reply:
Thanks a lot for the reply! I have a better understanding of it now and will go ahead and fix it. Thank you!